### PR TITLE
Update code-analysis.md

### LINKED ISSUE #28825: Add missing analyzers to table
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 657f3da7-ec96-7d56-f1b4-4787e0f0fd84
* Version Independent ID: 657f3da7-ec96-7d56-f1b4-4787e0f0fd84
* Content: [Code analysis in ASP.NET Core apps](https://learn.microsoft.com/en-us/aspnet/core/diagnostics/code-analysis?view=aspnetcore-8.0)
* Content Source: [aspnetcore/diagnostics/code-analysis.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/diagnostics/code-analysis.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

---
[Associated WorkItem - 81153](https://dev.azure.com/msft-skilling/Content/_workitems/edit/81153)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #28825

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/diagnostics/code-analysis.md](https://github.com/dotnet/AspNetCore.Docs/blob/e1026e42cf8fdb3ff713fbb50cdfde10c18d57de/aspnetcore/diagnostics/code-analysis.md) | [aspnetcore/diagnostics/code-analysis](https://review.learn.microsoft.com/en-us/aspnet/core/diagnostics/code-analysis?branch=pr-en-us-28835) |

### CODE REVIEW

File: aspnetcore/diagnostics/code-analysis.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ author: rick-anderson
 description: Learn about source code analysis in ASP.NET Core
 monikerRange: '>= aspnetcore-3.1'
 ms.author: riande
-ms.date: 11/22/2022
+ms.date: 5/2/2023
```
Comment (tdykstra) [inline aspnetcore/diagnostics/code-analysis.md:7]:
```suggestion
ms.date: 4/2/2023
```

File: aspnetcore/diagnostics/code-analysis.md [hunk 2]
```diff
@@ -32,6 +32,11 @@ uid: diagnostics/code-analysis
 | [ASP0017](xref:diagnostics/asp0017) | Non-breaking             | Invalid route pattern |
 | [ASP0018](xref:diagnostics/asp0018) | Non-breaking             | Unused route parameter |
 | [ASP0019](xref:diagnostics/asp0019) | Non-breaking             | Suggest using IHeaderDictionary.Append or the indexer |
+| [ASP0020](xref:diagnostics/asp0020) | Non-breaking             | Suggest using IHeaderDictionary.Append or the indexer |
+| [ASP0021](xref:diagnostics/asp0021) | Non-breaking             | Suggest using IHeaderDictionary.Append or the indexer |
+| [ASP0022](xref:diagnostics/asp0022) | Non-breaking             | Suggest using IHeaderDictionary.Append or the indexer |
+| [ASP0023](xref:diagnostics/asp0023) | Non-breaking             | Suggest using IHeaderDictionary.Append or the indexer |
+| [ASP0024](xref:diagnostics/asp0024) | Non-breaking             | Suggest using IHeaderDictionary.Append or the indexer |
```
Comment (tdykstra) [inline aspnetcore/diagnostics/code-analysis.md:39]: The description column needs to be updated for each diagnostic. But you might consider removing the table altogether and relying on the TOC for the list. This is a lot of duplication.

**Comment by Rick-Anderson** [comment]
The diff is terrible but you'll have to trust me.

**Comment by Rick-Anderson** [comment]
The last commit makes it easier to update.

**Review by tdykstra — APPROVED** [review]
If we keep this duplicate list it might make sense to show the error message for each diagnostic, since you don't get that in the TOC. This way it's just a duplicate, but I don't think it's a big deal, as I don't expect people will be looking in this list for the particular message, they'll search for it and get the specific file.

**Comment by tdykstra** [comment]
Note build warning -- ms.date too far in the future.

File: aspnetcore/diagnostics/code-analysis.md [hunk 3]
```diff
@@ -4,44 +4,51 @@ author: rick-anderson
 description: Learn about source code analysis in ASP.NET Core
 monikerRange: '>= aspnetcore-3.1'
 ms.author: riande
-ms.date: 11/22/2022
+ms.date: 5/2/2023
```
Comment (Rick-Anderson) [inline aspnetcore/diagnostics/code-analysis.md:7]:
```suggestion
ms.date: 4/2/2023
```

**Pull request merged**